lib/platform/bitbucket-server/index.js

```diff
@@ -611,18 +611,38 @@ async function updatePr(prNo, title, description) {
 
 async function mergePr(prNo) {
   logger.debug(`mergePr(${prNo})`);
-  // TODO: Needs implementation
   // Used for "automerge" feature
-  const { version } = (await api.get(
-    `./rest/api/1.0/projects/${config.projectKey}/repos/${
-      config.repositorySlug
-    }/pull-requests/${prNo}`
-  )).body;
+
+  const pr = await getPr(prNo);
+
+  // istanbul ignore if
+  if (!pr) {
+    logger.warn(`Failed to merge PR ${prNo} -- missing`);
+    return false;
+  }
+
+  // istanbul ignore if
+  if (pr.state != 'open') {
+    logger.warn(`Failed to merge PR ${prNo} -- already ${pr.state}`);
+    return false;
+  }
+
+  // istanbul ignore if
+  if (!pr.canMerge || pr.isConflicted) {
```

Comment:
This also shouldn't happen because the worker layer should check for it, but it's not harmful to add the extra check.